src/utils/components/DiskModal/UploadDiskModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { UPLOAD_STATUS } from '@kubevirt-utils/hooks/useCDIUpload/utils';
+import { getNamespace } from '@kubevirt-utils/resources/shared';
+import { kubevirtConsole } from '@kubevirt-utils/utils/utils';
+import { Form } from '@patternfly/react-core';
+import { isRunning } from '@virtualmachines/utils';
+
+import TabModal from '../TabModal/TabModal';
+
+import AdvancedSettings from './components/AdvancedSettings/AdvancedSettings';
+import BootSourceCheckbox from './components/BootSourceCheckbox/BootSourceCheckbox';
+import DiskInterfaceSelect from './components/DiskInterfaceSelect/DiskInterfaceSelect';
+import DiskNameInput from './components/DiskNameInput/DiskNameInput';
+import DiskSizeInput from './components/DiskSizeInput/DiskSizeInput';
+import DiskSourceUploadPVC from './components/DiskSourceSelect/components/DiskSourceUploadPVC/DiskSourceUploadPVC';
+import DiskTypeSelect from './components/DiskTypeSelect/DiskTypeSelect';
+import PendingChanges from './components/PendingChanges';
+import StorageClassAndPreallocation from './components/StorageClassAndPreallocation/StorageClassAndPreallocation';
+import { getDefaultCreateValues } from './utils/form';
+import { diskModalTitle, hotplugPromise } from './utils/helpers';
+import { addDisk, reorderBootDisk, uploadDataVolume } from './utils/submit';
+import { SourceTypes, V1DiskFormState, V1DiskModalProps } from './utils/types';
+
+const UploadDiskModal: FC<V1DiskModalProps> = ({
+  isOpen,
+  onClose,
+  onSubmit,
+  onUploadedDataVolume,
+  vm,
+}) => {
+  const { upload, uploadData } = useCDIUpload();
+  const isVMRunning = isRunning(vm);
+  const vmNamespace = getNamespace(vm);
+
+  const methods = useForm<V1DiskFormState>({
+    defaultValues: getDefaultCreateValues(vm, SourceTypes.UPLOAD),
+    mode: 'all',
+  });
+
+  const {
+    formState: { isSubmitting, isValid },
+    handleSubmit,
+  } = methods;
+
+  return (
+    <FormProvider {...methods}>
+      <TabModal
+        onClose={() => {
+          if (upload?.uploadStatus === UPLOAD_STATUS.UPLOADING) {
+            try {
+              upload.cancelUpload();
+            } catch (error) {
+              kubevirtConsole.error(error);
+            }
+          }
+          onClose();
+        }}
+        onSubmit={() =>
+          handleSubmit(async (data) => {
+            const uploadedDataVolume = await uploadDataVolume(vm, uploadData, data);
+
+            onUploadedDataVolume?.(uploadedDataVolume);
+
+            data.dataVolumeTemplate.spec.source.pvc = {
+              name: uploadedDataVolume.metadata.name,
+              namespace: uploadedDataVolume.metadata.namespace || vmNamespace,
+            };
+
+            const vmWithDisk = addDisk(data, vm);
+
+            const newVM = reorderBootDisk(vmWithDisk, data.disk.name, data.isBootSource, false);
+
+            return !isVMRunning ? onSubmit(newVM) : (hotplugPromise(newVM, data) as Promise<any>);
+          })()
```

Comment:
why are u running it?

Reply:
onSubmit= () => handleSubmit(handler)()

This because the TabModal onSubmit function types do not match with the handleSubmit returned function type